web/react/components/post_list.jsx

```diff
@@ -517,86 +517,88 @@ export default class PostList extends React.Component {
             var post = posts[order[i]];
             var parentPost = posts[post.parent_id];
 
-            var sameUser = false;
-            var sameRoot = false;
-            var hideProfilePic = false;
-            var prevPost = posts[order[i + 1]];
-
-            if (prevPost) {
-                sameUser = prevPost.user_id === post.user_id && post.create_at - prevPost.create_at <= 1000 * 60 * 5;
-
-                sameRoot = Utils.isComment(post) && (prevPost.id === post.root_id || prevPost.root_id === post.root_id);
-
-                // hide the profile pic if:
-                //     the previous post was made by the same user as the current post,
-                //     the previous post is not a comment,
-                //     the current post is not a comment,
-                //     the current post is not from a webhook
-                //     and the previous post is not from a webhook
-                if ((prevPost.user_id === post.user_id) &&
-                        !Utils.isComment(prevPost) &&
-                        !Utils.isComment(post) &&
-                        (!post.props || !post.props.from_webhook) &&
-                        (!prevPost.props || !prevPost.props.from_webhook)) {
-                    hideProfilePic = true;
+            if (!parentPost || parentPost.state !== Constants.POST_DELETED) {
```

Comment:
I think it would be better to not wrap all the following in this if statement. Can you move it up and do something like:

```
if (parentPost && parentPost.state === Constants.POST_DELETED) {
    continue;
}
```

Also the condition seems weird, it's basically we only display the post if we don't have it's parent (so a regular post or a reply that we don't have the parent for) or if it's not deleted. Maybe add a comment explaining why it's being done